Implement basic kick user functionality

OTCompa opened the pull request:
This commit implements the basic kick functionality based on the scope and requirements outline.
Usage is limited to users with kick permissions.

OTCompa committed:
implement basic kick functionality

OTCompa committed:
implement basic kick functionality

brtran4 commented on internal/discord/commands.go:
```diff
@@ -1,11 +1,36 @@
+		optionMap[opt.Name] = opt
+	}
+
+	userToKick := optionMap["user"].UserValue(nil).ID
```
let's add error checking to make sure that person is in the discord

brtran4 replied:
another thing to consider is that we'll be searching up users all the time, and we have to do error checking each time. we can move this to the handler file and just call that to decrease code duplication

brtran4 commented on internal/discord/commands.go:
```diff
@@ -1,11 +1,36 @@
+	var err error
+	if optionMap["reason"] != nil {
+		err = d.Session.GuildMemberDeleteWithReason(i.GuildID, userToKick, optionMap["reason"].StringValue())
+	} else {
+		err = d.Session.GuildMemberDelete(i.GuildID, userToKick)
+	}
```
we don't want to kick a user without a specific reason, so we should check that there is a reason. if there is not, we yell at the mod to give us a reason before kicking

brtran4 commented on internal/discord/commands.go:
```diff
@@ -1,11 +1,36 @@
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		StartInteraction(s, i.Interaction, tempstr)
+	} else {
+		tempstr := fmt.Sprintf("User <@%v> has been kicked.", userToKick)
+		fmt.Printf("%v\n", tempstr)
+		StartInteraction(s, i.Interaction, tempstr)
```
ultimately we would also want to DM the user why they are being kicked. not sure if we can do it now, but it would be nice if we're able to support that from the get-go

OTCompa replied:
How should we handle cases where the user has DMs turned off?

brtran4 replied:
ideally we should catch the error message then just ban them without giving them the reason. we would still have to log that to the channel

OTCompa committed:
Merge branch 'kick' of https://github.com/OTCompa/moddingway into kick

OTCompa committed:
return a message instead of kicking if no reason provided

OTCompa committed:
ensure person is in server before kicking

also added a function in command-handlers.go to check whether user is in server

OTCompa committed:
DM kicked user regarding kick

OTCompa committed:
Merge branch 'main' into kick

brtran4 commented on internal/discord/commands.go:
```diff
@@ -10,7 +12,75 @@ import (
+		StartInteraction(s, i.Interaction, tempstr)
+	} else {
+		tempstr := fmt.Sprintf("User <@%v> has been kicked.", userToKick)
+		fmt.Printf("%v\n", tempstr)
```
let's also make sure that we are logging the reason for the kick too. in the future we'll need to make an embed formatting which mod kicked who and why
